docs/src/reference/the-traversal.asciidoc

```diff
@@ -2302,6 +2302,281 @@ link:++https://tinkerpop.apache.org/javadocs/x.y.z/core/org/apache/tinkerpop/gre
 link:++https://tinkerpop.apache.org/javadocs/x.y.z/core/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.html#mean-org.apache.tinkerpop.gremlin.process.traversal.Scope-++[`mean(Scope)`],
 link:++https://tinkerpop.apache.org/javadocs/x.y.z/core/org/apache/tinkerpop/gremlin/process/traversal/Scope.html++[`Scope`]
 
+
+[[mergeedge-step]]
+=== MergeEdge Step
+
+The `mergeE()` -step is used to add edges and their properties to a graph in a "create
+if not exist" fashion. The `mergeE` step can also be used to find edges matching a given
+pattern. This step was added to Gremlin as part of the TinkerPop 3.6 release. Prior to
+that, it was necessary to write more complex queries using steps such as `coalesce`,
+`fold` and `unfold`, or injecting maps into queries and writing additional steps to
+manipulate them. The `mergeE` step makes these types of operation much simpler for the
+query writer. The input passed to `mergeE` can be either a map, or a child traversal that
+produces a map.
+
+NOTE: There is a corresponding `mergeV` step that can be used when creating vertices.
+
+Additionally, `optional` steps may be combined with `mergeE`, to take action depending on
+whether a vertex was created, or already existed. There are various ways that `mergeE` can
+be used. The simplest being to provide a single map of keys and values, along with the
+source and target vertex IDs, as a parameter.  An ID and a label may also be provided but
+this is optional. If the `T.id` or `T.label` are provided, they need to be enclosed in
+parenthesis. The `mergeE` step can be used directly from the GraphTraversalSource - `g`,
```

Comment:
> If the `T.id` or `T.label` are provided, they need to be enclosed in parenthesis.

This isn't true anymore after my follow-on commit to this one. I couldn't figure out a place to properly discuss `Map` initialization in the documentation. Perhaps it should just be included here and in mergev section as a `NOTE:`?

Reply:
hmm - i just realized our examples are being processed by Groovy and not the `gremlin-language`, so what you wrote is technically correct in that context. not sure how we want to rectify that now, but it needs to be clarified a bit.